src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1148,14 +1156,34 @@ public new Task<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior, C
         internal ManualResetValueTaskSource<NpgsqlConnector> ExecutionCompletion { get; }
             = new();
 
+        NpgsqlConnector? TryGetBoundConnector(NpgsqlConnection? connection)
+        {
+            if (_connector is not null)
+            {
+                Debug.Assert(_connection is null);
+                return _connector;
+            }
+
+            Debug.Assert(connection is not null);
+            if (connection.TryGetBoundConnector(out var connector))
+                return connector;
+
+            Debug.Assert(connection.Settings.Multiplexing);
+            return null;
+        }
+
         internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavior, bool async, CancellationToken cancellationToken)
         {
             var conn = CheckAndGetConnection();
             _behavior = behavior;
 
+            var connector = TryGetBoundConnector(conn);
+            if (conn is null && connector is not null && behavior.HasFlag(CommandBehavior.CloseConnection))
```

Comment:
tiny: reverse order to check HasFlag first: it will return false in the vast majority of cases, so we short-circuit and skip the other two checks.

Possibly further: since TryGetBoundConnector is only used here, we could just inline it here, and then we wouldn't have to repeat the check here outside.